validate/validate.go

```diff
@@ -0,0 +1,60 @@
+package validate
+
+import (
+	"os"
+	"regexp"
+)
+
+var envVarNameRegexp *regexp.Regexp = regexp.MustCompile(`(?m)^[A-Z0-9\_]+$`)
```

Comment:
I think lazy initialize this to prevent the regex always been compiled

Reply:
I would argue this is actually a better solution, as the one-time cost of compilation is negligible, and this provides startup-time confidence that the regex is valid, as it would otherwise immediately panic. If making it lazy, it would either be a potentially unpredictable panic when the function is first used, or we'd need to add a bunch of error handling and change the function signature to add error returns (or silently fail). For those reasons, I think this is a better option for maintenance safety, at a negligible cost of performance.

However, if you disagree and think lazy compilation is preferable, I'd be happy to implement that.